herddb-core/src/main/java/herddb/sql/CalcitePlanner.java

```diff
@@ -341,6 +350,80 @@ public TranslatedQuery translate(String defaultTableSpace, String query, List<Ob
         }
     }
 
+
+    private TranslatedQuery calculateShowCreateTable(String query, String defaultTablespace, List<Object> parameters) {
+        String items[] = {"SHOW","CREATE", "TABLE"};
+        if (Arrays.stream(items).allMatch(query::contains)) {
+            query = query.substring(Arrays.stream(items).collect(Collectors.joining(" ")).length()).trim();
+            String tableSpace = defaultTablespace;
+            String tableName;
+            if (query.contains(".")) {
+                String tokens[] = query.split("\\.");
+                tableSpace = tokens[0].trim();
+                tableName = tokens[1].trim();
+            } else {
+                tableName = query.trim();
+            }
+
+            TableSpaceManager tableSpaceManager = manager.getTableSpaceManager(tableSpace);
+
+
+            if (tableSpaceManager == null) {
+                throw new TableSpaceDoesNotExistException(String.format("Tablespace %s does not exist.", tableSpace));
+            }
+
+            AbstractTableManager tableManager = tableSpaceManager.getTableManager(tableName);
+
+            if (tableManager == null || tableManager.getCreatedInTransaction() > 0) {
+                throw new TableDoesNotExistException(String.format("Table %s does not exist.", tableName));
+            }
+
+
+            Table t  = tableManager.getTable();
+            if (t == null) {
+                throw new TableDoesNotExistException(String.format("Table %s does not exist.", tableName));
+            }
+
+            StringBuilder sb = new StringBuilder("CREATE TABLE "+tableSpace+"."+tableName);
```

Comment:
Please create an utility class for this part, that is the builder of the create table string.
This way you can do unit tests without starting a server.
Please also create unit tests for:
Multi column pk
Auto increment
Indexes, brin and hash
Table without pk (heap table)
All data types, even not null ones

Reply:
Great suggestion. I will refactor this.